### DISCUSSION TITLE
Add option to block messages containing spam

### DISCUSSION THREAD

**Discussion by rubo77** [element-hq/element-meta#416] [General]
### Description

There is a lot of spam coming from IRC bridges at the moment. most of them contain the same external link again and again.

### Solution
Please add a wordlist to Riot for each room, or in the Riot config file for systemwide Spam filtering.

If a message is filtered the "spammer" should get a response like "Your messageis was filtered because..."

**Comment by lampholder** [comment]
There's something of a philosophical question about this one. Spam filtering could be implemented at various layers:
- IRC
- IRC<-->Matrix Homeserver Bridge
- Matrix Homeserver
- Client

For almost all of those layers (all but IRC, actually), responding to the sender saying why their message has been filtered doesn't make a lot of sense. How would you reply? Are you telling them that a subset of native/bridged members didn't see the message because of filtering?

All this said, spam is annoying and bad. If it's all coming from the same user then you can choose in your Riot client to ignore them, but if it's lots of users spamming the same (or similar) messages then we don't currently have a client-configurable option for that.

It might be nice to have custom message-filtering options set on the user-account level, such that users can choose to suppress messages matching certain criteria, but this would need a lot of thought to be implemented in a way that both was easy to use and didn't give users a footgun to accidentally silence large segments of legitimate conversation.

**Reply by jevstev** [reply]
Why not implement it at the application level the same way that usenet did with the killfile?
https://en.wikipedia.org/wiki/Kill_file

**Comment by rubo77** [comment]
>in a way that both was easy to use and didn't give users a footgun to accidentally silence large segments of legitimate conversation.

We could limit the entries for the blacklist to sentences with a minimum of words or certain characters that have to be included (for example " http")

**Comment by aaronraimist** [comment]
I'm not sure why you need to prevent users from shooting themselves in the foot. I mean sure a future version of this feature could add a warning to say that a particular filter might block lots of unintended messages but everything from IRC clients to Twitter allows you to block any phrase you don't want to see, doesn't matter if it is spam or if it is a spoiler for a movie you haven't seen yet. Once this feature is added, I don't think Riot should prevent a user from blocking any phrase they don't want to see.

**Comment by MTRNord** [comment]
Not sure if it is covered by this issue but a ignore by keyword function is what I would want. As for example there might be a person that in general writes readworthy content but sometimes tries to promote a project that I dont want to hear about as it in my opinion is pure spam. (As for example being offtopic or for other reasons)

**Comment by aaronraimist** [comment]
Yeah I would say it is
> Please add a wordlist to Riot for each room, or in the Riot config file for systemwide Spam filtering.

This is related to vector-im/element-web#7521/#6819 where I want the ability to ignore something but also see that there is something there that I ignored.

I'm considering writing a spec proposal for this.
